ynr/apps/sopn_parsing/helpers/extract_pages.py

```diff
@@ -18,6 +21,73 @@ def extract_pages_for_ballot(ballot, manual_upload=False):
     :type ballot: candidates.models.Ballot
 
     """
+    if ballot.sopn.uploaded_file.path.endswith(".pdf"):
+        pass
+    elif ballot.sopn.uploaded_file.path.endswith(".html"):
+        print(
+            f"Warning! HTML file types are not supported; trying to convert {ballot.sopn.uploaded_file.path} to pdf"
+        )
+
+        html_file_path = ballot.sopn.uploaded_file.path
+
+        with tempfile.NamedTemporaryFile(suffix=".pdf") as temp_file:
+            # convert the doc and save to a temp file
+            pypandoc.convert_file(
+                html_file_path, to="pdf", outputfile=temp_file.name
+            )
+
+            # save the pdf file to the sopn document
+            pdf_file_name = ballot.sopn.uploaded_file.name.replace(
+                ".html", ".pdf"
+            )
+
+            ballot.sopn.uploaded_file.save(
+                name=pdf_file_name, content=temp_file, save=True
+            )
+            ballot.sopn.save()
+
+    elif ballot.sopn.uploaded_file.path.endswith(".docx"):
+        print(
+            f"Warning! docx file types are not supported; trying to convert {ballot.sopn.uploaded_file.path} to pdf"
+        )
+
+        docx_file_path = ballot.sopn.uploaded_file.path
+
+        with tempfile.NamedTemporaryFile(suffix=".pdf") as temp_file:
+            pypandoc.convert_file(
+                docx_file_path, to="pdf", outputfile=temp_file.name
+            )
+
+            pdf_file_name = ballot.sopn.uploaded_file.name.replace(
+                ".docx", ".pdf"
+            )
+
+            ballot.sopn.uploaded_file.save(
+                name=pdf_file_name, content=temp_file, save=True
+            )
+            ballot.sopn.save()
+    else:
+        raise ValueError("File type not currently supported")
+    # check if this is the only document with this source url and if so attempt
+    # some optimisations before we try and parse the page numbers
+    try:
```

Comment:
A lot of this was removed in a recent merge to master, have you rebased lately? As this will add this back in

Reply:
I have rebased, not sure what happened here. Will investigate.